### PR TITLE
test: achieve 100% code coverage and consolidate the test suite

### PR DESCRIPTION
**Pull request by konradmichalik**
## Summary

- Reach **100% code coverage** (classes, methods, lines) across the package — up from ~95% lines / ~85% methods / ~61% classes
- Add focused tests for previously uncovered real logic (CLI renderer severity sorting & verbose detailed output, validator multi-issue branch, mismatch value handling, encoding control-character detection, HTML class-attribute diffs, empty-XLIFF handling, config auto-detection edge cases, recursive collector path safety, generic prefix detection)
- Exclude genuinely unreachable defensive guards from coverage via `@codeCoverageIgnore` (e.g. `file_get_contents()`/`getcwd()`/`realpath()`/`preg_split()`/`glob()` returning `false`, `class_exists()` fallbacks, exhaustive-`match` default arms)
- Aggressively consolidate the suite: merge near-identical tests into `#[DataProvider]`s, remove trivial getter/enum tests, delete two redundant duplicate test files, and fold per-validator metadata checks into one parametrized test
- Fix PHPUnit 13 deprecations/notices (drop `expects($this->any())`, use stubs instead of unexpected mocks)

Net result: **795 tests pass with zero deprecations/notices/warnings**, ~660 fewer lines of test code, lint and PHPStan clean.

## Changes

- `src/**` (14 files) — `@codeCoverageIgnore` annotations on unreachable defensive branches only (no behavioral changes)
- `tests/src/Validator/ValidatorMetadataTest.php` — new parametrized test covering `getShortName`/`resultTypeOnValidationFailure`/`supportsParser` for all validators
- `tests/src/**` (41 files) — new branch-coverage tests, data-provider consolidation, removed redundant tests
- `tests/src/PluginExtendedTest.php`, `tests/src/Utility/PathUtilityExtendedTest.php` — deleted (1:1 duplicates of base tests)

## Summary by CodeRabbit

* **New Features**
  * Added support for loading configuration from a project’s `composer.json` when a custom config file is referenced.
  * Expanded validator metadata checks to keep supported parser and result-type behavior consistent.

* **Bug Fixes**
  * Improved handling of edge cases in file detection, parsing, path normalization, and schema validation.
  * Added broader validation for empty inputs, missing content, and mixed file formats.

* **Tests**
  * Streamlined and expanded automated test coverage across configuration, parsing, result rendering, and validator behavior.

### CODE REVIEW

File: tests/src/Fixtures/config/unsupported.txt [hunk 1]
```diff
@@ -1 +1 @@
-some content
+some content
```
[the remark this answers is not in the document]
Comment (konradmichalik) [reply tests/src/Fixtures/config/unsupported.txt]: @coderabbitai Fixed — added the trailing newline to the fixture (verified with xxd; EditorConfig/CGL now pass).

File: tests/src/Parser/XliffParserTest.php [hunk 2]
```diff
@@ -633,6 +633,30 @@ public function testGetContentByKeyFallsBackToTargetWhenSourceIsEmptyXliff2(): v
...
+        $this->assertSame([], $parser->extractKeys());
+    }
+
+    public function testGetContentByKeyReturnsNullWhenNoTransUnits(): void
+    {
+        $file = $this->tempDir.'/no_units_content.xlf';
+        file_put_contents($file, <<<'EOT'
+<?xml version="1.0" encoding="utf-8"?>
+<xliff version="1.2"><file source-language="en" datatype="plaintext" original="x"><body></body></file></xliff>
+EOT);
```
[the remark this answers is not in the document]
Comment (konradmichalik) [reply tests/src/Parser/XliffParserTest.php:654]: @coderabbitai Done — both empty-XLIFF fixtures now declare the default xliff 1.2 namespace (xmlns="urn:oasis:names:tc:xliff:document:1.2"), matching real .xlf files. The no-trans-units branch stays covered (tests green).

File: tests/src/Result/ValidationResultCliRendererTest.php [hunk 3]
```diff
@@ -323,6 +326,57 @@ public function testRenderWarningInVerboseMode(): void
...
+            excludePatterns: [],
+            recursive: false,
+            fileDetector: new PrefixFileDetector(),
+            validators: ValidatorRegistry::getAvailableValidators(),
+            config: new TranslationValidatorConfig(),
+        );
+
+        self::assertInstanceOf(ValidationResult::class, $result);
+
+        return $result;
```
[the remark this answers is not in the document]
Comment (konradmichalik) [reply tests/src/Result/ValidationResultCliRendererTest.php:380]: @coderabbitai Tightened — buildRealValidationResult() now runs only the validators this scenario needs (Mismatch=ERROR, HtmlTag=WARNING, XliffSchema), so the severity ordering is deterministic and unaffected by other validators/files. All getValidatorSeverity branches remain covered.

File: tests/src/Validator/KeyNamingConventionValidatorTest.php [hunk 4]
```diff
@@ -533,10 +574,8 @@ public function testDetectKeyConventionsReturnsMixedConventions(): void
 
     public function testConfigurationLoadingWithInvalidConvention(): void
     {
-        $config = $this->createMock(\MoveElevator\ComposerTranslationValidator\Config\TranslationValidatorConfig::class);
-        $config->expects($this->any())
-            ->method('getValidatorSettings')
-            ->with('KeyNamingConventionValidator')
+        $config = $this->createStub(\MoveElevator\ComposerTranslationValidator\Config\TranslationValidatorConfig::class);
+        $config->method('getValidatorSettings')
             ->willReturn(['convention' => 'invalid_convention']);
```
[the remark this answers is not in the document]
Comment (konradmichalik) [reply tests/src/Validator/KeyNamingConventionValidatorTest.php]: @coderabbitai Good catch — restored the contract: both tests now use createMock with expects($this->once())->method('getValidatorSettings')->with('KeyNamingConventionValidator'). This keeps the suite deprecation/notice-free (expects()+with() is fine in PHPUnit 13) while asserting the correct settings key.
